File: infrastructure/control/osimctrl/src/start-opensim.py
#!/usr/bin/python

import os.path
import re
import subprocess
import sys

### CONFIGURE THESE PATHS ###
binaryPath = "/home/opensim/opensim/opensim-current/bin"
pidPath = "/tmp/OpenSim.pid"
### END OF CONFIG ###

if os.path.exists(pidPath):
  print >> sys.stderr, "ERROR: OpenSim PID file %s still present.  Assuming OpenSim has been started already." % pidPath
  sys.exit(1)

# If PID isn't set then we'll check the screen list.  
# However, this is a much less perfect mechanism since OpenSimulator may have been started outside screen
screenList = ""

try:
  screenList = subprocess.check_output("screen -list", shell=True)
except:
  None

if re.match("\s+\d+\.OpenSim", screenList):
  print >> sys.stderr, "ERROR: Screen session for OpenSim already started."
  sys.exit(1)

File: infrastructure/control/osimctrl/src/start-opensim.pl
#!/usr/bin/perl -w

### CONFIGURE THESE PATHS ###
my $binaryPath = "/home/opensim/opensim/opensim-current/bin";
my $pidPath = "/tmp/OpenSim.pid";
### END OF CONFIG ###

-f $pidPath and die "ERROR: OpenSim PID file $pidPath still present.  Assuming OpenSim has been started already.\n";

# If PID isn't set then we'll check the screen list.  
# However, this is a much less perfect mechanism since OpenSimulator may have been started outside screen
`screen -list` =~ /OpenSim/ and die "ERROR: Screen session for OpenSim already started.\n";

chdir($binaryPath) or die;
execCmd("screen -S OpenSim -d -m mono --debug OpenSim.exe");

# I would like to perform this check but it generates false positives and I would rather not put in a messy sleep
# unless (-f $pidPath) { warn "WARNING: OpenSimulator PID file $pidPath not found after startup.  Set this in [Startup] PIDFile in OpenSim.ini"; }

if (`screen -list` =~ /OpenSim/)
{
    print "OpenSim starting in screen instance..\n";
}
else
{
    die "ERROR: OpenSim did not start.\n";
}

sub execCmd
{
    my $cmd = $_[0];
    print "Excecuting command $cmd\n";
    system ($cmd) && die;
}
